File: go-sdk/pkg/models/operations/getfollowers.go
package operations

type GetFollowersQueryParams struct {
	Page    *int32  `queryParam:"style=form,explode=true,name=page"`
	PerPage *int32  `queryParam:"style=form,explode=true,name=per_page"`
	Sort    *string `queryParam:"style=form,explode=true,name=sort"`
}

// GetFollowers200ApplicationJSON
// A follower
type GetFollowers200ApplicationJSON struct {
	ID           *int32  `json:"id,omitempty"`
	Name         *string `json:"name,omitempty"`
	Path         *string `json:"path,omitempty"`
	ProfileImage *string `json:"profile_image,omitempty"`
	TypeOf       *string `json:"type_of,omitempty"`
	UserID       *int32  `json:"user_id,omitempty"`
}

type GetFollowersRequest struct {
	QueryParams GetFollowersQueryParams
}

type GetFollowersResponse struct {
	ContentType                           string
	StatusCode                            int64
	GetFollowers200ApplicationJSONObjects []GetFollowers200ApplicationJSON
}

File: ts-sdk/src/sdk/models/operations/getfollowers.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



export class GetFollowersQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=page" })
  page?: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=per_page" })
  perPage?: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=sort" })
  sort?: string;
}


// GetFollowers200ApplicationJson
/** 
 * A follower
**/
export class GetFollowers200ApplicationJson extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=id" })
  id?: number;

  @SpeakeasyMetadata({ data: "json, name=name" })
  name?: string;

  @SpeakeasyMetadata({ data: "json, name=path" })
  path?: string;

  @SpeakeasyMetadata({ data: "json, name=profile_image" })
  profileImage?: string;

  @SpeakeasyMetadata({ data: "json, name=type_of" })
  typeOf?: string;

  @SpeakeasyMetadata({ data: "json, name=user_id" })
  userId?: number;
}


export class GetFollowersRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: GetFollowersQueryParams;
}


export class GetFollowersResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata({ elemType: GetFollowers200ApplicationJson })
  getFollowers200ApplicationJSONObjects?: GetFollowers200ApplicationJson[];
}
